core/commands/object.go

```diff
@@ -580,19 +586,59 @@ func rmLinkCaller(req cmds.Request, root *dag.Node) (key.Key, error) {
 		return "", err
 	}
 
-	name := req.Arguments()[2]
+	path := strings.Split(req.Arguments()[2], "/")
 
-	err = root.RemoveNodeLink(name)
+	nnode, err := rmLink(req.Context(), nd.DAG, root, path)
 	if err != nil {
 		return "", err
 	}
 
-	newkey, err := nd.DAG.Add(root)
+	return nnode.Key()
+}
+
+func rmLink(ctx context.Context, ds dag.DAGService, root *dag.Node, path []string) (*dag.Node, error) {
+	if len(path) == 1 {
+		// base case, remove node in question
+		err := root.RemoveNodeLink(path[0])
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = ds.Add(root)
```

Comment:
this is just making sure the object is in `ds`? not sure why it's needed, in which case won't it be?